src/modules/Storage.rb

```diff
@@ -7126,6 +7125,42 @@ def GetDmraidToMdadm()
       return mapping
     end
 
+    # Updates /boot partition information
+    #
+    # It relies on SpecialBootHandling to determine what changes need to be done.
+    # To write changes in the current map, it calls SetPartitionFormat, SetPartitionId
+    # and SetPartitionMount methods, as SetTargetMap does not work properly.
+    #
+    # @param [Hash] old_tg Original target map (usually GetTargetMap())
+    # @see SpecialBootHandling
+    # @see SetPartitionFormat
+    # @see SetPartitionId
+    # @see SetPartitionMount
+    def update_boot_partition_data(old_tg)
+      new_tg = SpecialBootHandling(old_tg)
+      new_tg.each do |dev, disk|
+        old_partitions = old_tg.fetch(dev, {}).fetch("partitions")
+        disk["partitions"].each_with_index do |part, idx|
+          old_part = old_partitions[idx]
+
+          # format and fstype
+          if old_part["format"] != part["format"] || old_part["fstype"] != part["fstype"]
+            SetPartitionFormat(part["device"], part["format"], part["fstype"])
+          end
+
+          # fsid
+          if old_part["fsid"] != part["fsid"]
+            SetPartitionId(part["device"], part["fsid"])
+          end
+
+          # mount point
+          if old_part["mount"] != part["mount"]
+            SetPartitionMount(part["device"], part["mount"])
+          end
+        end
```

Comment:
As long as only /boot partition is supposed to exist, I guess we could break the loop when we found it.